tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -1465,24 +1470,61 @@ void CorProfiler::InternalAddInstrumentation(WCHAR* id, CallTargetDefinition* it
 
         std::scoped_lock<std::mutex> moduleLock(module_ids_lock_);
 
-        definitions_ids_.emplace(definitionsId);
+        if (enable)
+        {
+            definitions_ids_.emplace(definitionsId);
+        }
+        else
+        {
+            definitions_ids_.erase(definitionsId);
+        }
 
-        Logger::Info("Total number of modules to analyze: ", module_ids_.size());
-        if (rejit_handler != nullptr)
+        if (enable)
         {
-            std::promise<ULONG> promise;
-            std::future<ULONG> future = promise.get_future();
-            tracer_integration_preprocessor->EnqueueRequestRejitForLoadedModules(module_ids_, integrationDefinitions, &promise);
+            integration_definitions_.reserve(integration_definitions_.size() + integrationDefinitions.size());
+            for (const auto& integration : integrationDefinitions)
+            {
+                integration_definitions_.push_back(integration);
+            }
 
-            // wait and get the value from the future<int>
-            const auto& numReJITs = future.get();
-            Logger::Debug("Total number of ReJIT Requested: ", numReJITs);
-        }
+            Logger::Info("Total number of modules to analyze: ", module_ids_.size());
+            if (rejit_handler != nullptr)
+            {
+                std::promise<ULONG> promise;
+                std::future<ULONG> future = promise.get_future();
+                tracer_integration_preprocessor->EnqueueRequestRejitForLoadedModules(module_ids_,
+                                                                                     integrationDefinitions, &promise);
 
-        integration_definitions_.reserve(integration_definitions_.size() + integrationDefinitions.size());
-        for (const auto& integration : integrationDefinitions)
+                // wait and get the value from the future<int>
+                const auto& numReJITs = future.get();
+                Logger::Debug("Total number of ReJIT Requested: ", numReJITs);
+            }
+        }
+        else
         {
-            integration_definitions_.push_back(integration);
+            // remove the call target definitions
+            std::vector<IntegrationDefinition> integration_definitions = integration_definitions_;
+            integration_definitions_.clear();
```

Comment:
It doesn't look like we keep track of which integration definitions belong to a definitionsId. Have we tested that turning off the security feature DOES NOT turn off other automatic instrumentation integrations?

Reply:
Yes, indeed. Only requested CallTargets are restored / rejitted

Reply:
I got confused the first time reading this code but now it makes sense to me. Thanks!